pandas/core/indexes/base.py

```diff
@@ -5226,7 +5226,10 @@ def __getitem__(self, key):
             #  takes 166 µs + 2.1 ms and cuts the ndarray.__getitem__
             #  time below from 3.8 ms to 496 µs
             # if we already have ndarray[bool], the overhead is 1.4 µs or .25%
-            key = np.asarray(key, dtype=bool)
+            if is_extension_array_dtype(getattr(key, "dtype", None)):
```

Comment:
clearer or more performant to do `isinstance(key, ExtensionArray)`?

Reply:
Thought about this too, but does not work unfortunately, since key is a Series here